samcli/local/docker/lambda_image.py

```diff
@@ -197,7 +197,13 @@ def _build_image(self, base_image, docker_tag, layers, is_debug_go, stream=None)
             for layer in layers:
                 tar_paths[layer.codeuri] = "/" + layer.name
 
-            with create_tarball(tar_paths) as tarballfile:
+            # Set tar file permission to 777
+            # This is need for systems without unix like permission bits(Windows) while creating a unix image
+            def set_item_permission(tar_info):
+                tar_info.mode = 777
+                return tar_info
+
+            with create_tarball(tar_paths, tar_filter=set_item_permission) as tarballfile:
```

Comment:
Should we do this only for Windows? I was just wondering if this introduces any issues with other operating systems. Can we verify with other OS?

Reply:
Updated to only set in Windows.